airflow/www/templates/airflow/dag.html

```diff
@@ -521,7 +521,7 @@ <h4 class="modal-title" id="dagModalLabel">
     });
 
     $('#btn_edit_dagrun').click(function() {
-      window.location = "{{ url_for('DagModelView.show', pk=dag.dag_id) }}";
+      window.location = "{{ url_for('DagModelView.edit', pk=dag.dag_id) }}";
```

Comment:
Not a comment on this PR, but why is this in JS - couldn't it just be a normal HTML link.

(JS that could be written as plain HTML is a bug bear of mine)

Reply:
I think this could work.
```html
<form action="/dagmodel/edit/{{ dag.dag_id }}">
  <button id="btn_edit_dagrun" type="submit" class="btn btn-primary">
              Edit
  </button>
</form>
```
instead of this
```html
<button id="btn_edit_dagrun" type="button" class="btn btn-primary">
            Edit
</button>
```

Reply:
@feng-tao @XD-DENG @verdan What do you guys think? Should we change the links generated in JS to HTML links where possible?

Reply:
The advantage of using `url_for('DagModelView.show', pk=dag.dag_id)` is that we don't have to worry about the url generation and if flask, for some reason, changes this url from `/dagmodel/show/<pk>` to `/dagmodel/show?<pk>` we do not need to change anything.

Reply:
@ashb As @verdan says he is working on this change. So should we get that into `master` so he can rebase onto it and change this as well?